utils.py

```diff
@@ -291,3 +290,142 @@ def plot_mean_node_arity(self, show_counts=False):
         )
         if show_counts:
             plt.bar_label(bars, fmt="{:,.0f}")
+
+    def calc_mutations_per_tree(self):
+        tree_mutations = np.zeros(self.ts.num_trees)
+        for tree in self.ts.trees():
+            tree_mutations[tree.index] = tree.num_mutations
+        return tree_mutations
+
+    def plot_mutations_per_tree(self, max_num_muts=None, show_counts=False):
+        fig, ax = plt.subplots()
+        bins = None
+        if max_num_muts is not None:
+            bins = range(max_num_muts + 1)
+            trees_with_many_muts = np.sum(
+                self.calc_mutations_per_tree() > max_num_muts)
+            ax.text(
+                0.01,
+                0.99,
+                f"there are {trees_with_many_muts:,} trees\nwith more than {max_num_muts:,} mutations",
+                transform=ax.transAxes,
+            )
+        counts, edges, bars = plt.hist(
+            self.calc_mutations_per_tree(), bins=bins, edgecolor="black"
```

Comment:
As it is expensive, there's no need to call `self.calc_mutations_per_tree` twice here and on 306.